### PR TITLE
Demonstrate actual pagination API

### PR DESCRIPTION
**Pull request by DanielSiepmann**
The paginator itself can not be used to create a pagination.
Instead a wrapper is needed which will provide proper API.

Backport: 10.4

### CODE REVIEW

File: Documentation/ApiOverview/Pagination/Index.rst [hunk 1]
```diff
@@ -43,4 +43,11 @@ Code-Example for the :php:`ArrayPaginator`:
    $paginator->getKeyOfFirstPaginatedItem(); // returns 5
    $paginator->getKeyOfLastPaginatedItem(); // returns 5
    $paginator->getAllPageNumbers(); // returns [1,2,3]
```
Comment (mbrodala) [inline Documentation/ApiOverview/Pagination/Index.rst:45]: This method does not exist actually. Can you remove this, while we're at it?

File: Documentation/ApiOverview/Pagination/Index.rst [hunk 2]
```diff
@@ -43,4 +43,11 @@ Code-Example for the :php:`ArrayPaginator`:
...
+   
+   // use TYPO3\CMS\Extbase\Pagination\QueryResultPaginator;
+   $pagination = new SimplePagination($paginator);
+   $pagination->getAllPageNumbers(); // returns 3
```
Comment (mbrodala) [inline Documentation/ApiOverview/Pagination/Index.rst:49]:
```suggestion
   $pagination->getAllPageNumbers(); // returns [1, 2, 3]
```

**Review by mbrodala — CHANGES_REQUESTED** [review]

**Comment by DanielSiepmann** [comment]
Relates to https://review.typo3.org/c/Packages/TYPO3.CMS/+/69723/

**Review by mbrodala — APPROVED** [review]

**Comment by mbrodala** [comment]
Since this is an essential change I'll merge it like this. If you want to also add a Fluid example using the pagination, please add another change.

**Pull request merged**